<h1>ACTIVITY 1 - PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3>Analyst: Mark Ernest Garay</h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h3>Converting Data Set(s) into Data Frame</h3>

<h2>Requirements</h2>
<ul>
    <li>
        Name of the Data Frame
    </li>
    <li>
        File name
    </li>
</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [4]:
penguins.describe()


bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h2>Identifying Null Values</h2>

In [5]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>Filtering</h2>

In [7]:
penguins['sex']

0        male
1      female
2      female
3         NaN
4      female
        ...  
339      male
340    female
341      male
342      male
343    female
Name: sex, Length: 344, dtype: object

In [8]:
penguins['sex'].value_counts()

sex
male      168
female    165
Name: count, dtype: int64

In [9]:
penguins['sex'] == "male"

0       True
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341     True
342     True
343    False
Name: sex, Length: 344, dtype: bool

In [10]:
penguins[penguins['sex'] == "male"]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
5       Adelie  Torgersen            39.3           20.6              190.0   
7       Adelie  Torgersen            39.2           19.6              195.0   
13      Adelie  Torgersen            38.6           21.2              191.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
..         ...        ...             ...            ...                ...   
334  Chinstrap      Dream            50.2           18.8              202.0   
336  Chinstrap      Dream            51.9           19.5              206.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
0         3750.0  male  2007  
5         3650.0  male  2007  
7         4675.0  male  2007  
13        3800.0  male  2007  
14        4400.0  male  2007  
..           ...   ...   ...  
334       3800.0  male  2009  
336       3950.0  male  2009  
339       4000.0  male  2009  
341       3775.0  male  2009  
342       4100.0  male  2009  

[168 rows x 8 columns]

In [11]:
malepenguins =penguins[penguins['sex'] == "male"]

In [12]:
malepenguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
5       Adelie  Torgersen            39.3           20.6              190.0   
7       Adelie  Torgersen            39.2           19.6              195.0   
13      Adelie  Torgersen            38.6           21.2              191.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
..         ...        ...             ...            ...                ...   
334  Chinstrap      Dream            50.2           18.8              202.0   
336  Chinstrap      Dream            51.9           19.5              206.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
0         3750.0  male  2007  
5         3650.0  male  2007  
7         4675.0  male  2007  
13        3800.0  male  2007  
14        4400.0  male  2007  
..           ...   ...   ...  
334       3800.0  male  2009  
336       3950.0  male  2009  
339       4000.0  male  2009  
341       3775.0  male  2009  
342       4100.0  male  2009  

[168 rows x 8 columns]

In [13]:
adelie_penguins = penguins[penguins['species'] == "Adelie"]

In [14]:
adelie_penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [15]:
island_penguins = penguins[penguins['island'] == "Dream"]

In [16]:
island_penguins

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [17]:
penguins['flipper_length_mm'] >= 200

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341    False
342     True
343    False
Name: flipper_length_mm, Length: 344, dtype: bool

In [18]:
(penguins['flipper_length_mm'] >= 200) & (penguins['species'] == 'Adelie' )

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [19]:
adelielength = penguins[(penguins['flipper_length_mm'] >= 200) & (penguins['species'] == 'Adelie')]

In [20]:
adelielength

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
53   Adelie     Biscoe            42.0           19.5              200.0   
90   Adelie      Dream            35.7           18.0              202.0   
91   Adelie      Dream            41.1           18.1              205.0   
95   Adelie      Dream            40.8           18.9              208.0   
101  Adelie     Biscoe            41.0           20.0              203.0   
123  Adelie  Torgersen            41.4           18.5              202.0   
129  Adelie  Torgersen            44.1           18.0              210.0   
137  Adelie      Dream            40.2           20.1              200.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
53        4050.0    male  2008  
90        3550.0  female  2008  
91        4300.0    male  2008  
95        4300.0    male  2008  
101       4725.0    male  2009  
123       3875.0    male  2009  
129       4000.0    male  2009  
137       3975.0    male  2009  
151       4000.0    male  2009

In [21]:
adelielength = penguins[(penguins['flipper_length_mm'] >= 200) & (penguins['species'] == 'Adelie') & (penguins['body_mass_g'] <= 4000)]

In [22]:
adelielength

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
90   Adelie      Dream            35.7           18.0              202.0   
123  Adelie  Torgersen            41.4           18.5              202.0   
129  Adelie  Torgersen            44.1           18.0              210.0   
137  Adelie      Dream            40.2           20.1              200.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
90        3550.0  female  2008  
123       3875.0    male  2009  
129       4000.0    male  2009  
137       3975.0    male  2009  
151       4000.0    male  2009

<h2>Filling up Null Values</h2>

In [23]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [24]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean()) 

In [25]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [26]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean()) 

In [27]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [28]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean()) 

In [29]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [30]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0]) 

In [31]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       2
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>Changing the Data Types</h2>

In [32]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [33]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('object')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     342 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(3), float64(4), object(1)
memory usage: 15.0+ KB


In [35]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [36]:
penguins['flipper_inch'] = np.round(penguins['flipper_length_mm'] / 25.4 , 1)

In [37]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000 , 1)

In [38]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1       18.70000         181.000000   
1       Adelie  Torgersen            39.5       17.40000         186.000000   
2       Adelie  Torgersen            40.3       18.00000         195.000000   
3       Adelie  Torgersen             NaN       17.15117         200.915205   
4       Adelie  Torgersen            36.7       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8       19.80000         207.000000   
340  Chinstrap      Dream            43.5       18.10000         202.000000   
341  Chinstrap      Dream            49.6       18.20000         193.000000   
342  Chinstrap      Dream            50.8       19.00000         210.000000   
343  Chinstrap      Dream            50.2       18.70000         198.000000   

     body_mass_g     sex  year  flipper_inch  mass_kg  
0    3750.000000    male  2007           7.1      3.8  
1    3800.000000  female  2007           7.3      3.8  
2    3250.000000  female  2007           7.7      3.2  
3    4201.754386    male  2007           7.9      4.2  
4    3450.000000  female  2007           7.6      3.4  
..           ...     ...   ...           ...      ...  
339  4000.000000    male  2009           8.1      4.0  
340  3400.000000  female  2009           8.0      3.4  
341  3775.000000    male  2009           7.6      3.8  
342  4100.000000    male  2009           8.3      4.1  
343  3775.000000  female  2009           7.8      3.8  

[344 rows x 10 columns]

<h2>Exploratory Data Analytics (EDA)</h2>

<h2>Question 1</h2>

<h2> Which specie has the highest average flipper length?</h2>

In [39]:
adelie_l =  penguins[penguins['species'] == "Adelie"]

In [40]:
adelie_l['flipper_length_mm'].mean()

np.float64(190.02575792551556)

In [41]:
chinstrap_l =  penguins[penguins['species'] == "Chinstrap"]

In [42]:
chinstrap_l['flipper_length_mm'].mean()

np.float64(195.8235294117647)

In [43]:
gentoo_l =  penguins[penguins['species'] == "Gentoo"]

In [44]:
gentoo_l['flipper_length_mm'].mean()

np.float64(217.05576777966422)

<h2>Insight No. 1</h2>

<h4>Base on the data, Gentoo has the longest average flipper length</h4>

<h2>Question 2</h2>

<h2>What is the average body mass of each penguins species?</h2>

In [45]:
adelie_m =  penguins[penguins['species'] == "Adelie"]

In [46]:
adelie_m['mass_kg'].mean()

np.float64(3.7052631578947373)

In [47]:
chinstrap_m =  penguins[penguins['species'] == "Chinstrap"]

In [48]:
chinstrap_m['mass_kg'].mean()

np.float64(3.730882352941177)

In [49]:
gentoo_m =  penguins[penguins['species'] == "Gentoo"]

In [50]:
gentoo_m['mass_kg'].mean()

np.float64(5.069354838709677)

<h2>Insight No. 2</h2>

<h4>Adelie has the average mass of 3.70 kg, Chinstrap has 3.73 kg and Gentoo has 5.06 kg  </h4>

<h2>Question 3</h2>

<h2>Which specie has the highest average bill length?</h2>

In [51]:
adelie_b =  penguins[penguins['species'] == "Adelie"]

In [52]:
adelie_m['bill_length_mm'].mean()

np.float64(38.79139072847682)

In [53]:
chinstrap_b =  penguins[penguins['species'] == "Chinstrap"]

In [54]:
chinstrap_b['bill_length_mm'].mean()

np.float64(48.83382352941177)

In [55]:
gentoo_b =  penguins[penguins['species'] == "Gentoo"]

In [56]:
gentoo_b['bill_length_mm'].mean()

np.float64(47.50487804878048)

<H2>Insight No. 3</H2>

<h4>We can see that the specie that has the highest average bill length is the Chinstrap</h4>

<h2>Question 4</h2>

<h2>Which island has the most number of Adelie?</h2>

In [57]:
penguins[penguins['species'] == 'Adelie'] ['island'].describe()

count       152
unique        3
top       Dream
freq         56
Name: island, dtype: object

<h2>Insight No. 4</h2>

<h4>The Adelie penguins has the most number in the island of Dream</h4>

<h2>Question 5</h2>

<h2>Which specie are mostly female?</h2>

In [58]:
penguins[penguins['sex'] == 'female'] ['species'].describe()

count        165
unique         3
top       Adelie
freq          73
Name: species, dtype: object

<h2>Insight No. 5</h2>

<h4>Adelie penguins are mostly female</h4>

<h2>Question 6</h2>

<h2>Is the body mass of a penguin related to it's sex?</h2>

In [59]:
penguins[penguins['sex'] == 'female'] ['mass_kg'].describe()

count    165.000000
mean       3.861212
std        0.669327
min        2.700000
25%        3.400000
50%        3.600000
75%        4.600000
max        5.200000
Name: mass_kg, dtype: float64

In [60]:
penguins[penguins['sex'] == 'male'] ['mass_kg'].describe()

count    179.000000
mean       4.516201
std        0.784767
min        3.000000
25%        3.900000
50%        4.300000
75%        5.300000
max        6.300000
Name: mass_kg, dtype: float64

<h2>Insight No. 6</h2>

<h4>The male penguins are shown heavier than the females</h4>

<h2>Question 7</h2>

<h2> Which specie has the highest average bill depth?</h2>

In [61]:
adelie_bd =  penguins[penguins['species'] == "Adelie"]

In [62]:
np.round(adelie_bd['bill_depth_mm'].mean() ,1 )

np.float64(18.3)

In [63]:
chinstrap_bd =  penguins[penguins['species'] == "Chinstrap"]

In [64]:
np.round(chinstrap_bd['bill_depth_mm'].mean(), 1)

np.float64(18.4)

In [65]:
gentoo_bd =  penguins[penguins['species'] == "Gentoo"]

In [66]:
np.round(gentoo_bd['bill_depth_mm'].mean(), 1)

np.float64(15.0)

<h2>Insight No. 7</h2>

<h4>After rounding off the mean, we can see that the specie that has the highest average bill depth is the Chinstrap</h4>

In [67]:
penguins.head(10)

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1       18.70000         181.000000   
1  Adelie  Torgersen            39.5       17.40000         186.000000   
2  Adelie  Torgersen            40.3       18.00000         195.000000   
3  Adelie  Torgersen             NaN       17.15117         200.915205   
4  Adelie  Torgersen            36.7       19.30000         193.000000   
5  Adelie  Torgersen            39.3       20.60000         190.000000   
6  Adelie  Torgersen            38.9       17.80000         181.000000   
7  Adelie  Torgersen            39.2       19.60000         195.000000   
8  Adelie  Torgersen            34.1       18.10000         193.000000   
9  Adelie  Torgersen            42.0       20.20000         190.000000   

   body_mass_g     sex  year  flipper_inch  mass_kg  
0  3750.000000    male  2007           7.1      3.8  
1  3800.000000  female  2007           7.3      3.8  
2  3250.000000  female  2007           7.7      3.2  
3  4201.754386    male  2007           7.9      4.2  
4  3450.000000  female  2007           7.6      3.4  
5  3650.000000    male  2007           7.5      3.6  
6  3625.000000  female  2007           7.1      3.6  
7  4675.000000    male  2007           7.7      4.7  
8  3475.000000    male  2007           7.6      3.5  
9  4250.000000    male  2007           7.5      4.2

In [68]:
penguins.tail(10)

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
334  Chinstrap  Dream            50.2           18.8              202.0   
335  Chinstrap  Dream            45.6           19.4              194.0   
336  Chinstrap  Dream            51.9           19.5              206.0   
337  Chinstrap  Dream            46.8           16.5              189.0   
338  Chinstrap  Dream            45.7           17.0              195.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  flipper_inch  mass_kg  
334       3800.0    male  2009           8.0      3.8  
335       3525.0  female  2009           7.6      3.5  
336       3950.0    male  2009           8.1      4.0  
337       3650.0  female  2009           7.4      3.6  
338       3650.0  female  2009           7.7      3.6  
339       4000.0    male  2009           8.1      4.0  
340       3400.0  female  2009           8.0      3.4  
341       3775.0    male  2009           7.6      3.8  
342       4100.0    male  2009           8.3      4.1  
343       3775.0  female  2009           7.8      3.8

<h2>Data Aggregation</h2>

In [69]:
penguins_per_island = penguins.groupby(['species', 'island']).size()

In [70]:
penguins_per_island

species    island   
Adelie     Biscoe        44
           Dream         56
           Torgersen     52
Chinstrap  Biscoe         0
           Dream         68
           Torgersen      0
Gentoo     Biscoe       124
           Dream          0
           Torgersen      0
dtype: int64

In [71]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1       18.70000         181.000000   
1       Adelie  Torgersen            39.5       17.40000         186.000000   
2       Adelie  Torgersen            40.3       18.00000         195.000000   
3       Adelie  Torgersen             NaN       17.15117         200.915205   
4       Adelie  Torgersen            36.7       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8       19.80000         207.000000   
340  Chinstrap      Dream            43.5       18.10000         202.000000   
341  Chinstrap      Dream            49.6       18.20000         193.000000   
342  Chinstrap      Dream            50.8       19.00000         210.000000   
343  Chinstrap      Dream            50.2       18.70000         198.000000   

     body_mass_g     sex  year  flipper_inch  mass_kg  
0    3750.000000    male  2007           7.1      3.8  
1    3800.000000  female  2007           7.3      3.8  
2    3250.000000  female  2007           7.7      3.2  
3    4201.754386    male  2007           7.9      4.2  
4    3450.000000  female  2007           7.6      3.4  
..           ...     ...   ...           ...      ...  
339  4000.000000    male  2009           8.1      4.0  
340  3400.000000  female  2009           8.0      3.4  
341  3775.000000    male  2009           7.6      3.8  
342  4100.000000    male  2009           8.3      4.1  
343  3775.000000  female  2009           7.8      3.8  

[344 rows x 10 columns]

In [72]:
penguins_fl = penguins.groupby(['species', 'island']) ['mass_kg'].mean()

In [73]:
penguins_fl

species    island   
Adelie     Biscoe       3.715909
           Dream        3.691071
           Torgersen    3.711538
Chinstrap  Biscoe            NaN
           Dream        3.730882
           Torgersen         NaN
Gentoo     Biscoe       5.069355
           Dream             NaN
           Torgersen         NaN
Name: mass_kg, dtype: float64

<h2>Correlation Analysis</h2>

<strong> <hr><strong>

<h3>Testing correlation using Pearson R</h3>

<hr>

In [74]:
bill_length_mm = penguins['bill_depth_mm'].corr(penguins['bill_length_mm'])

In [75]:
bill_length_mm

np.float64(-0.23505287035553254)

In [76]:
from scipy.stats import f_oneway

In [77]:
penguins[penguins['sex'] == 'male'] ['body_mass_g']

0      3750.000000
3      4201.754386
5      3650.000000
7      4675.000000
8      3475.000000
          ...     
334    3800.000000
336    3950.000000
339    4000.000000
341    3775.000000
342    4100.000000
Name: body_mass_g, Length: 179, dtype: float64

In [78]:
n_use, p_value = f_oneway(penguins[penguins['sex'] == 'male'] ['body_mass_g'], penguins[penguins['sex'] == 'female'] ['body_mass_g'])

In [79]:
p_value

np.float64(3.0226600487050246e-15)

<h2>Question 8</h2>

<h2>Which specie has the longest bill length in the island of Dream?</h2>

In [80]:
penguins_bl = penguins.groupby(['species'] , ['island'] == 'Dream') ['bill_length_mm'].mean()

In [81]:
penguins_bl

species
Adelie       38.791391
Chinstrap    48.833824
Gentoo       47.504878
Name: bill_length_mm, dtype: float64

<h2>Insight No. 8 </h2>

<h4>Based on the data, Chinstrap has the longest average bill length in the island of Dream</h4>

In [82]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1       18.70000         181.000000   
1       Adelie  Torgersen            39.5       17.40000         186.000000   
2       Adelie  Torgersen            40.3       18.00000         195.000000   
3       Adelie  Torgersen             NaN       17.15117         200.915205   
4       Adelie  Torgersen            36.7       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8       19.80000         207.000000   
340  Chinstrap      Dream            43.5       18.10000         202.000000   
341  Chinstrap      Dream            49.6       18.20000         193.000000   
342  Chinstrap      Dream            50.8       19.00000         210.000000   
343  Chinstrap      Dream            50.2       18.70000         198.000000   

     body_mass_g     sex  year  flipper_inch  mass_kg  
0    3750.000000    male  2007           7.1      3.8  
1    3800.000000  female  2007           7.3      3.8  
2    3250.000000  female  2007           7.7      3.2  
3    4201.754386    male  2007           7.9      4.2  
4    3450.000000  female  2007           7.6      3.4  
..           ...     ...   ...           ...      ...  
339  4000.000000    male  2009           8.1      4.0  
340  3400.000000  female  2009           8.0      3.4  
341  3775.000000    male  2009           7.6      3.8  
342  4100.000000    male  2009           8.3      4.1  
343  3775.000000  female  2009           7.8      3.8  

[344 rows x 10 columns]

<h2>Question 9</h2>

<h2>Does the bill depth of a penguin correlates to it's body mass?</h2>

In [83]:
bill_depth_mm = penguins['body_mass_g'].corr(penguins['bill_depth_mm'])

In [84]:
bill_depth_mm

np.float64(-0.47191562118606645)

<h2>Insight No. 9</h2>

<h4>We can see that there is no correlation between the bill depth and the body mass of a penguin</h4>

<h2>Question 10</h2>

<h2>What sex of a penguin has the highest number on each island?</h2>

In [85]:
penguins_sex_island = penguins.groupby(['island', 'sex']).size()

In [86]:
penguins_sex_island

island     sex   
Biscoe     female    80
           male      88
Dream      female    61
           male      63
Torgersen  female    24
           male      28
dtype: int64

<h2>Insight No. 10</h2>

<h4>The data shows that the number of male penguins is higher than the number of female penguins on each island</h4>

<h2>Question 11</h2>

<h2>How does the average flipper length vary between the different islands?</h2>

In [87]:
fl_per_island = penguins.groupby('island')['flipper_length_mm'].mean()

In [88]:
fl_per_island

island
Biscoe       209.654257
Dream        193.072581
Torgersen    191.382985
Name: flipper_length_mm, dtype: float64

<h2>Insight No. 11</h2>

<h4>Biscoe Island has the highest average flipper length among the islands, other than that the average flipper length of penguins vary slightly across the two remaining</h4>

<h2>Question 12</h2>

<h2>Is there a correlation between the bill length and the body mass of a penguin?</h2>

In [89]:
body_mass_g = penguins['bill_length_mm'].corr(penguins['body_mass_g'])

In [90]:
body_mass_g

np.float64(0.5951098244376303)

<h2>Insight No. 12</h2>

<h4>Base on the data, there is a correlation between the bill length and the body mass of a penguin.</h4>

<h2>Question 13</h2>

<h2>Which specie has the highest average body mass in the island of Biscoe?</h2>

In [93]:
penguins_bm = penguins.groupby(['species'] , ['island'] == 'Dream') ['body_mass_g'].mean()

In [94]:
penguins_bm

species
Adelie       3703.958910
Chinstrap    3733.088235
Gentoo       5068.965761
Name: body_mass_g, dtype: float64

<h2>Insight No. 13</h2>

<h4>The data shows that Gentoo has the highest average body mass in the island of Biscoe.</h4>